feat(lua): make :lua =expr print result of expr (#16912)
neovim/neovim: runtime/doc/lua.txt

                                                        *:lua*
:[range]lua {chunk}
                        Executes Lua chunk {chunk}.

>                         if {chunk} starts with "=" the result of the
>                         expression in chunk gets printed.So `:lua =expr`
>                         works same as `:lua print(vim.inspect(expr))`

Reviewer: ```suggestion
                        if {chunk} starts with "=" the rest of the chunk is
                        evaluated as an expression and printed. `:lua =expr`
                        is equivalent to `:lua print(vim.inspect(expr))`
```
